src/web/sitemap.rs

```diff
@@ -31,36 +49,42 @@ pub fn sitemap_handler(req: &mut Request) -> IronResult<Response> {
         })
         .collect::<Vec<(String, String)>>();
 
-    let mut resp = ctry!(Page::new(releases).to_resp("sitemap"));
-    resp.headers
-        .set(ContentType("application/xml".parse().unwrap()));
-    Ok(resp)
+    SitemapXml { releases }.into_response()
 }
 
 pub fn robots_txt_handler(_: &mut Request) -> IronResult<Response> {
     let mut resp = Response::with("Sitemap: https://docs.rs/sitemap.xml");
-    resp.headers.set(ContentType("text/plain".parse().unwrap()));
+    resp.headers.set(ContentType::plaintext());
+
     Ok(resp)
 }
 
-pub fn about_handler(req: &mut Request) -> IronResult<Response> {
-    let mut content = BTreeMap::new();
+#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
+pub(crate) struct About {
+    /// The current version of rustc that docs.rs is using to build crates
+    pub rustc_version: Option<String>,
+    /// The default crate build limits
+    pub limits: Limits,
+}
+
+impl_webpage!(About = "docsrs/about.html");
 
+pub fn about_handler(req: &mut Request) -> IronResult<Response> {
     let conn = extension!(req, Pool).get()?;
     let res = ctry!(conn.query("SELECT value FROM config WHERE name = 'rustc_version'", &[]));
 
+    let mut rustc_version = None;
     if let Some(row) = res.iter().next() {
         if let Some(Ok::<Value, _>(res)) = row.get_opt(0) {
             if let Some(vers) = res.as_str() {
-                content.insert("rustc_version".to_string(), Value::String(vers.to_string()));
+                rustc_version = Some(vers.to_owned());
```

Comment:
This copy doesn't look necessary.
```suggestion
            if let Some(Value::String(vers)) = res {
                rustc_version = Some(vers);
```